components/blitz/src/omero/gateway/facility/TablesFacility.java

```diff
@@ -0,0 +1,526 @@
+                    }
+                    dataArray[i] = rowData;
+                    types[i] = ROIData.class;
+                }
+            }
+            return new TableData(header, descriptions, types, dataArray);
+        } catch (Exception e) {
+            handleException(this, e, "Could not load table data");
+        }
+        return null;
+    }
+
+    /**
+     * Get all available tables for a the specified object
+     * 
+     * @param ctx
+     *            The {@link SecurityContext}
+     * @param parent
+     *            The {@link DataObject}
+     * @return See above
+     * @throws DSOutOfServiceException
+     *             If the connection is broken, or not logged in
+     * @throws DSAccessException
+     *             If an error occurred while trying to retrieve data from OMERO
+     *             service.
+     */
+    public Collection<FileData> getAvailableTables(SecurityContext ctx,
+            DataObject parent) throws DSOutOfServiceException,
+            DSAccessException {
+        Collection<FileData> result = new ArrayList<FileData>();
+        try {
+            MetadataFacility mf = gateway.getFacility(MetadataFacility.class);
+            List<AnnotationData> annos = mf.getAnnotations(ctx, parent,
+                    Arrays.asList(new Class[] { FileAnnotationData.class }),
+                    null);
+            for (AnnotationData anno : annos) {
+                FileAnnotationData fanno = (FileAnnotationData) anno;
+                if (fanno.isTablesFile())
+                    result.add(new FileData(((FileAnnotation) fanno
+                            .asAnnotation()).getFile()));
+            }
+        } catch (Exception e) {
+            handleException(this, e, "Could not load tables");
+        }
+        return result;
+    }
+
+    /**
+     * Create a {@link Column} with the specified data
+     * 
+     * @param header
+     *            The header (column name)
+     * @param description
+     *            Description
+     * @param type
+     *            The type of data
+     * @param data
+     *            The data
+     * @return The {@link Column}
+     */
+    private Column createColumn(String header, String description,
+            Class<?> type, Object[] data) {
+        Column c = null;
+
+        if (type.equals(Boolean.class)) {
+            boolean[] d = new boolean[data.length];
+            for (int i = 0; i < data.length; i++)
+                d[i] = (Boolean) data[i];
+            c = new BoolColumn(header, description, d);
+        }
+
+        if (type.equals(Double[].class)) {
+            double[][] d = new double[data.length][];
+            int l = 0;
+            for (int i = 0; i < data.length; i++) {
+                Double[] src = (Double[]) data[i];
+                double[] dst = new double[src.length];
+                for (int j = 0; j < src.length; j++)
+                    dst[j] = src[j];
```

Comment:
Apache Commons has tons of useful array type stuff like https://commons.apache.org/proper/commons-lang/javadocs/api-2.6/org/apache/commons/lang/ArrayUtils.html#toPrimitive%28java.lang.Double[]%29